internal/testrunner/runners/system/runner.go

```diff
@@ -620,20 +620,26 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 	// (TODO in future) Optionally exercise service to generate load.
 	logger.Debug("checking for expected data in data stream...")
 	var hits *hits
+	oldHits := 0
 	passed, err := waitUntilTrue(func() (bool, error) {
 		if signal.SIGINT() {
 			return true, errors.New("SIGINT: cancel waiting for policy assigned")
 		}
 
 		var err error
 		hits, err = r.getDocs(dataStream)
+		if hits.size() == 0 {
+			return false, err
+		}
 
-		if config.Assert.HitCount > 0 {
-			return hits.size() >= config.Assert.HitCount, err
+		ret := hits.size() == oldHits
```

Comment:
This is going to make all tests to wait till timeout or till the count of documents is stable, isn't it? Could we continue taking into acount if `config.Assert.HitCount` is configured so we don't extend the wait for the rest of cases?

Maybe to solve this issue it is enough with adding a sleep when `hits.size() >= config.Assert.HitCount`?

Reply:
Breaking this down into scenarios :

### HitCount configured
- Number of docs less than the count in `n` iterations - Timeout happens as the `true` condition is never satisfied - Test fails
- Number of docs equal in 2 successive iterations with a gap of 5 seconds - No more  docs to ingest and then verify if the count is matching - Catching false positives
- Number of docs never equal - Some issue in the test case causing the docs ingestion to go into infinite loop - Timeout and Test fails

### HitCount Not configured
- Docs ingestion complete - Meaning the `hits.size()` is same in 2 consecutive iterations with 1 second each - Adds up 1 second on each test case but provides us with data if we were to validate something else.
- Docs ingestion incomplete - Just picking up those docs which are ingested in the first iteration and validating against them

I think it should not add up a lot of time for each run , as I see the test suites are running in parallel , but we shall have the data ingested for any other validation - Just to avoid testing against incomplete data - WDYT?